config/runtime.exs

```diff
@@ -88,27 +88,43 @@ if config_env() == :prod do
       """)
   end
 
-  # hCaptcha
-  hcaptcha_site_key = System.get_env("HCAPTCHA_SITE_KEY")
-  hcaptcha_secret_key = System.get_env("HCAPTCHA_SECRET_KEY")
-  hcaptcha_url = System.get_env("HCAPTCHA_URL")
+  # Captcha
+  captcha_site_key = System.get_env("CAPTCHA_SITE_KEY") || System.get_env("HCAPTCHA_SITE_KEY")
 
-  if hcaptcha_site_key not in [nil, ""] and hcaptcha_secret_key not in [nil, ""] do
-    config =
-      [secret_key: hcaptcha_secret_key, site_key: hcaptcha_site_key]
-      |> put_if_not_empty.(:url, hcaptcha_url)
+  captcha_secret_key =
+    System.get_env("CAPTCHA_SECRET_KEY") || System.get_env("HCAPTCHA_SECRET_KEY")
+
+  captcha_url = System.get_env("CAPTCHA_URL") || System.get_env("HCAPTCHA_URL")
+
+  if captcha_site_key not in [nil, ""] and captcha_secret_key not in [nil, ""] do
+    captcha_type = if captcha_site_key =~ "-", do: "hcaptcha", else: "friendlycaptcha"
+    Logger.info("Using the #{captcha_type} captcha provider")
 
-    config :keila, KeilaWeb.Hcaptcha, config
+    default_captcha_url =
+      if captcha_type == "hcaptcha",
+        do: "https://hcaptcha.com/siteverify",
+        else: "https://api.friendlycaptcha.com/api/v1/siteverify"
+
+    config =
+      [
+        secret_key: captcha_secret_key,
+        site_key: captcha_site_key,
+        url: default_captcha_url,
+        type: captcha_type
+      ]
+      |> put_if_not_empty.(:url, captcha_url)
+
+    config :keila, KeilaWeb.captcha(), config
   else
     Logger.warn("""
-    hCaptcha not configured.
+    Captcha not configured.
     Keila will fall back to using hCaptcha’s staging configuration.
 
-    To configure hCaptcha, use the following environment variables:
+    To configure a captcha, use the following environment variables:
 
-    - HCAPTCHA_SITE_KEY
-    - HCAPTCHA_SECRET_KEY
-    - HCAPTCHA_URL (defaults to https://hcaptcha.com/siteverify)
+    - CAPTCHA_SITE_KEY (or HCAPTCHA_SITE_KEY for compatibility purposes)
```

Comment:
I think we can drop the reference to the `HCAPTCHA` env variables here because we don’t want to encourage their use anymore.